analyses/cell-type-wilms-tumor-14/scripts/summary_results.R

```diff
@@ -89,7 +88,25 @@ arg_df <- tidyr::expand_grid(
 dfs <- purrr::pmap(arg_df,
              \(anchor_assay, level) concat_sample(anchor_assay, level, meta)
 )
+# add rough tumor cell classification, CM = tumor cells, immune/endothelium = normal
+normal_compartments <- c("immune", "endothelium")
 final_table <- dfs %>%
-  purrr::reduce(left_join)
+  purrr::reduce(left_join) %>%
+  mutate( tumor_cell_classification_RNA = case_when(
+    cell_type_assignment_RNA_compartment %in% normal_compartments ~ "normal",
+    cell_type_assignment_RNA_compartment == "fetal_nephron" & cell_type_assignment_RNA_celltype == "Cap mesenchyme" ~ "tumor",
+    .default = "unknown"
+  ) ) %>%
+  { if ("SCT" %in% assays) mutate(., tumor_cell_classification_SCT = case_when(
+    cell_type_assignment_SCT_compartment %in% normal_compartments ~ "normal",
+    cell_type_assignment_SCT_compartment == "fetal_nephron" & cell_type_assignment_SCT_celltype == "Cap mesenchyme" ~ "tumor",
+    .default = "unknown"
+  ) ) else . } %>% 
+  # use logNormalize as "default" final result, combine immune, endo, and stroma compartments
+  mutate( cell_type_assignment = if_else(cell_type_assignment_RNA_compartment != "fetal_nephron", 
+                                         cell_type_assignment_RNA_compartment, 
+                                         cell_type_assignment_RNA_celltype ) ) %>%
+  mutate( tumor_cell_classification = tumor_cell_classification_RNA )
```

Comment:
While what you have does work, I find that nesting in `if` statements is likely to lead to confusion, so I am breaking up the pipeline a bit for clarity. 

```suggestion
  ) )

if ("SCT" %in% assays) {
  final_table <- final_table %>%
    mutate( tumor_cell_classification_SCT = case_when(
      cell_type_assignment_SCT_compartment %in% normal_compartments ~ "normal",
      cell_type_assignment_SCT_compartment == "fetal_nephron" & cell_type_assignment_SCT_celltype == "Cap mesenchyme" ~ "tumor",
      .default = "unknown"
    ) )
}

final_table <- final_table %>%
  # use logNormalize as "default" final result, combine immune, endo, and stroma compartments
  mutate( cell_type_assignment = if_else(cell_type_assignment_RNA_compartment != "fetal_nephron", 
                                         cell_type_assignment_RNA_compartment, 
                                         cell_type_assignment_RNA_celltype ) ) %>%
  mutate( tumor_cell_classification = tumor_cell_classification_RNA )
```